source/setup.Rmd

```diff
@@ -1,282 +1,288 @@
-or terminal (MacOS and Ubuntu) and press enter:
-
+Run JupyterLab Desktop using the command
 ```
-jupyter labextension install @techrah/text-shortcuts
+jlab
 ```
+Next, in the JupyterLab Desktop graphical interface that appears (Figure \@ref(fig:setup-jlab-gui)), 
+you will see text at the bottom saying "Python environment not found". Click "Install using the bundled installer"
+to set up the environment.
 
-### R, R packages, and the IRkernel 
-
-To have the software \index{R installation} used in this book available to you in JupyterLab, 
-you will need to install the R programming language,
-several R packages,
-and the \index{kernel!installation} IRkernel.
-To install versions of these that are compatible with the accompanying worksheets, 
-type the command shown below into the Anaconda Prompt (Windows) 
-or terminal (MacOS and Ubuntu). 
+```{r setup-jlab-gui, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "The JupyterLab Desktop graphical user interface.", out.width="85%", fig.align = "center", fig.retina = 2}
+knitr::include_graphics("img/setup/jlab-1.png")
+```
 
+Next, we need to add the JupyterLab Git extension (so that
+we can use version control directly from within JupyterLab Desktop),
+the IRkernel (to enable the R programming language),
+and various R software packages. Click "New session..." in the JupyterLab Desktop
+user interface, then scroll to the bottom, and click "Terminal" under the "Other" heading (Figure \@ref(fig:setup-jlab-gui-2)).
+```{r setup-jlab-gui-2, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "A JupyterLab Desktop session, showing the Terminal option at the bottom.", out.width="85%", fig.align = "center", fig.retina = 2}
+knitr::include_graphics("img/setup/jlab-2.png")
 ```
+In this terminal, run the following commands:
+```
+pip install --upgrade jupyterlab-git
 conda env update --file https://raw.githubusercontent.com/UBC-DSCI/data-science-a-first-intro-worksheets/main/environment.yml
 ```
-
-This command installs the specific R and package versions specified in
+The second command installs the specific R and package versions specified in
 the `environment.yml` file found in 
 [the worksheets repository](https://worksheets.datasciencebook.ca).
 We will always keep the versions in the `environment.yml` file updated
 so that they are compatible with the exercise worksheets that accompany the book.
 
 > You can also install the *latest* version of R 
-> and the R packages used in this book by typing the commands shown below 
-> in the Anaconda Prompt (Windows) 
-> or terminal (MacOS and Ubuntu) and pressing enter.
+> and the R packages used in this book by instead typing the commands shown below 
 > **Be careful though:** this may install package versions that are
 > incompatible with the worksheets that accompany the book; the automated
 > exercise feedback might tell you your answers are not correct even though
 > they are!
```

Comment:
Could we fix the package versions to what's in the docker container to avoid this?

Reply:
So, the `environment.yml` file is the one people should use to get precisely the right versions of things. I might actually just remove this note box to avoid the issue entirely.